src/GafferOSL/OSLShader.cpp

```diff
@@ -319,6 +333,20 @@ Plug *loadCompoundNumericParameter( const OSLQuery::Parameter *parameter, const
 	/// \todo Get from metadata
 	ValueType minValue( Imath::limits<BaseType>::min() );
 	ValueType maxValue( Imath::limits<BaseType>::max() );
+	if( metadata )
+	{
+		typedef typename TypedParameter<ValueType>::ObjectType ValueDataType;
```

Comment:
This was my suggestion, I'd (possibly incorrectly) assumed `TypedData<ValueType>` wouldn't give you the right result for the `GeometricTypedData` and would give the weirdo `V3fDataBase` stuff instead.